internal/uvm/scsi.go

```diff
@@ -323,9 +347,41 @@ func (uvm *UtilityVM) addSCSIActual(ctx context.Context, hostPath, uvmPath, atta
 	}
 
 	if err := uvm.modify(ctx, SCSIModification); err != nil {
-		return nil, fmt.Errorf("uvm::AddSCSI: failed to modify utility VM configuration: %s", err)
+		return fmt.Errorf("uvm::AddSCSI: failed to modify utility VM configuration: %s", err)
 	}
-	return sm, nil
+	return nil
+}
+
+// allocateSCSIDevice grants vm access to hostpath and increments the ref count of an existing scsi
+// device or allocates a new one if not already present.
+// Returns the resulting *SCSIMount, a bool indicating if the scsi device was already present,
+// and error if any.
+func (uvm *UtilityVM) allocateSCSIDevice(ctx context.Context, hostPath, uvmPath string, vmAccess VMAccessType) (*SCSIMount, bool, error) {
```

Comment:
I think `allocateSCSIMount` is a clearer name.

Reply:
And we can rename the deallocation function to match.

Reply:
Ah thank you, I couldn't decide on a good name for this :)